repositories/MerchStoreRepository.ts

```diff
@@ -53,11 +54,15 @@ export class MerchItemRepository extends BaseRepository<MerchandiseItemModel> {
     return this.repository.save(item);
   }
 
-  public async updateMerchItemInCollection(collection: string, hidden: boolean): Promise<void> {
+  public async updateMerchItemsInCollection(collection: string, changes: any): Promise<void> {
```

Comment:
We know `changes` should only contain fields from `MerchandiseItemModel` so we can declare it as `changes: Partial<MerchandiseItemModel>`. All the other methods should do something similar with `changes: Partial<T>`.

Reply:
@michl1001 what Sumeet's saying is essentially to change the type of `changes` from `any` to `Partial<MerchandiseItemModel>` since we know the type of it. Can you take care of this here?